--- miniupnpc/igd_desc_parse.c
+++ miniupnpc/igd_desc_parse.c
@@ -81,6 +81,11 @@
 		dstmember = datas->tmp.scpdurl;
 /*	else if( !strcmp(datas->cureltname, "deviceType") )
 		dstmember = datas->devicetype_tmp;*/
+	else if (!strcmp(datas->cureltname, "friendlyName") && l && !datas->friendlyNameParsed)
+	{
+		dstmember = datas->friendlyName;
+		datas->friendlyNameParsed = 1;
+	}
 	if(dstmember)
 	{
 		if(l>=MINIUPNPC_URL_MAXSIZE)
--- miniupnpc/igd_desc_parse.h
+++ miniupnpc/igd_desc_parse.h
@@ -37,6 +37,8 @@
 	struct IGDdatas_service IPv6FC;
 	/* tmp */
 	struct IGDdatas_service tmp;
+	char friendlyName[MINIUPNPC_URL_MAXSIZE];
+	int friendlyNameParsed;
 };
 
 void IGDstartelt(void *, const char *, int);
